spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -46,6 +47,8 @@ public class ExpireSnapshotsProcedure extends BaseProcedure {
 
   private static final Logger LOG = LoggerFactory.getLogger(ExpireSnapshotsProcedure.class);
 
+  private ExecutorService executorService = null;
```

Comment:
There is already `executorService` variable in `BaseProcedure` and `modifyIcebergTable()` must already close that executor service if it has been initialized.

Reply:
Thanks, I noticed. We are still using Spark 3.1. There is no logic like closing the threads pool in Spark 3.1. If necessary, I can open a PR to backport the relevant logic to Spark 3.1

Reply:
We are still using Spark 3.1 now. And I see the Remove Orphan Procedure and Expire Snapshots Procedure in Spark 3.1 don't have the code to turn off the thread pool, if necessary, I can open a new PR to backport this logic to Spark 3.1.